### PR TITLE
Update from Selenium 2 to Selenium 3 #313

### PR DESCRIPTION
**Pull request by btkawamotosn**
Please review #313 .

### CODE REVIEW

File: common/src/test/java/org/terasoluna/gfw/tutorial/selenium/FirefoxDriverPrepare.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    protected String userName;
+
+    @Value("${selenium.proxyUserPassword}")
+    protected String userPassword;
+
+    public void geckodriverSetup() {
+        if (System.getProperty("webdriver.gecko.driver") == null) {
+            FirefoxDriverManager.getInstance().version(geckodriverVersion)
+                    .forceCache().proxy(proxyHttpServer).proxyUser(userName)
+                    .proxyPass(userPassword).useTaobaoMirror().setup();
```
Comment (bttanakaysd) [inline common/src/test/java/org/terasoluna/gfw/tutorial/selenium/FirefoxDriverPrepare.java]: - [ ] Why did you add `useTaobaoMirror()`?

File: common/src/test/java/org/terasoluna/gfw/tutorial/selenium/WaitWebDriverEventListener.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            wait.until(
+                    (ExpectedCondition<Boolean>) wd -> ((JavascriptExecutor) wd)
+                            .executeScript("return document.readyState").equals(
+                                    "loading"));
+            wait.until(
+                    (ExpectedCondition<Boolean>) wd -> ((JavascriptExecutor) wd)
+                            .executeScript("return document.readyState").equals(
+                                    "complete"));
+        } catch (TimeoutException e) {
+            logger.debug("loading is not done");
```
Comment (bttanakaysd) [inline common/src/test/java/org/terasoluna/gfw/tutorial/selenium/WaitWebDriverEventListener.java]: - [ ] To notify the user, the log level must be at least WARN.

File: session-tutorial-complete/src/session-tutorial-complete-selenium/src/test/java/com/example/session/selenium/session/SessionTutorialTest.java [hunk 3]
```diff
@@ -204,18 +204,17 @@ public void testShowGoods() {
...
-            assertThat(driver.findElement(By.id("Swan Lake")).getText(), is(
-                    "Swan Lake"));
+            assertThat(driver.findElement(By.xpath("//tbody/tr[4]/td[1]"))
+                    .getText(), is("Swan Lake"));
         }
 
         // click the item name
         {
-            driver.findElement(By.id("Swan Lake")).click();
+            driver.findElement(By.linkText("Swan Lake")).click();
```
Comment (bttanakaysd) [inline session-tutorial-complete/src/session-tutorial-complete-selenium/src/test/java/com/example/session/selenium/session/SessionTutorialTest.java]: - [ ] Does `linkText ()` work correctly?

File: common/src/test/java/org/terasoluna/gfw/tutorial/selenium/FirefoxDriverPrepare.java (hunk 1 above)
Comment (btiwahorisu) [reply common/src/test/java/org/terasoluna/gfw/tutorial/selenium/FirefoxDriverPrepare.java]: Use `useTaobaoMirror()` to avoid the problem of the following link.
https://qiita.com/kazuki43zoo/items/71f4c09e644b45a4002c#github-api%E4%BD%BF%E7%94%A8%E6%99%82%E3%81%AB%E7%99%BA%E7%94%9F%E3%81%99%E3%82%8B403%E3%82%A8%E3%83%A9%E3%83%BC%E3%81%AE%E5%9B%9E%E9%81%BF%E6%96%B9%E6%B3%95

When running the test with travis, geckodriver's DL is done multiple in order to run multiple tests in parallel.
For this reason, it may get caught by the number of API calls and a 403 error may occur.
In order to avoid this problem, I avoided restriction on the number of API calls by doing DL from a mirror site that is not github.

File: common/src/test/java/org/terasoluna/gfw/tutorial/selenium/WaitWebDriverEventListener.java (hunk 2 above)
Comment (btiwahorisu) [reply common/src/test/java/org/terasoluna/gfw/tutorial/selenium/WaitWebDriverEventListener.java]: fix log level. e21f289

File: session-tutorial-complete/src/session-tutorial-complete-selenium/src/test/java/com/example/session/selenium/session/SessionTutorialTest.java (hunk 3 above)
Comment (btiwahorisu) [reply session-tutorial-complete/src/session-tutorial-complete-selenium/src/test/java/com/example/session/selenium/session/SessionTutorialTest.java]: `linkText ()` works correctly because html displays the string without line breaks.
